### DISCUSSION TITLE
low validation mean iou.

### DISCUSSION THREAD

**Discussion by PurvangL** [horovod/horovod#3735] [General]
My project contains Horovod, Dali as data preprocessing and tensorflow.fit for running training loop. I divide training and validation data for each worker so each gets it's own data and for each worker, DaliDataset object is created and preprocessing steps (normalization) is done on respective data  and pass to fit function assuming fit function will do accumulation the results across the shards. I use MetricAverageCallback as well from horovod. Tensorboard result shows all worker output instead of averaged output and also not sure since I have custom dataset, should I shard data based on worker id. For me, my val accuracy for same dataset coming 10% less compared to other mode of training (

(hvd + Dali + tf.fit) = val_meaniou - 20%
(hvd + tfdataset + tf.fit) = val_meaniou - 28%
(Dali + tf.fit) = val_meaniou - 29%
(tfdataset + tf.fit) = val_meaniou - 29%
)
any example for 1st case or any suggestion to improve result?
Thank you
